src/azure/AzureClientUnexpectedError.test.ts

```diff
@@ -0,0 +1,69 @@
+import { createTestLogger } from "@jupiterone/jupiter-managed-integration-sdk";
+import nock from "nock";
+import AzureClient from "./AzureClient";
+
+const CLIENT_ID =
+  process.env.AZURE_CLOUD_LOCAL_EXECUTION_CLIENT_ID || "example_token";
+const CLIENT_SECRET =
+  process.env.AZURE_CLOUD_LOCAL_EXECUTION_CLIENT_SECRET || "example_secret";
+const DIRRECTORY_ID =
+  process.env.AZURE_CLOUD_LOCAL_EXECUTION_DIRRECTORY_ID || "example_dirrectory";
+
+const logger = createTestLogger();
+
+jest.mock("node-fetch", () => {
+  return jest.fn().mockImplementation((url: string) => {
+    throw new Error();
+  });
+});
+
+describe("AzureClient fetch data with unknown error", () => {
+  beforeAll(() => {
+    nock.back.fixtures = `${__dirname}/../../test/fixtures/`;
+    process.env.CI
+      ? nock.back.setMode("lockdown")
+      : nock.back.setMode("record");
+  });
+
+  async function getClient() {
+    const azure = new AzureClient(
+      CLIENT_ID,
+      CLIENT_SECRET,
+      DIRRECTORY_ID,
+      logger,
+    );
+
+    return azure;
+  }
+
+  test("fetchUsers with a error", async () => {
+    const client = await getClient();
+    try {
+      await client.fetchUsers();
+    } catch (e) {
+      expect(e instanceof Error).toBe(true);
```

Comment:
I'd recommend checking the expected type of the error.
```suggestion
      expect(e).toInstanceOf(IntegrationError);
```
Or you may like (I think this should work):
```suggestion
      await expect(client.fetchUsers()).rejects.toBeInstanceOf(IntegrationError);
```